bam/reader.go

```diff
@@ -296,7 +296,9 @@ func (i *Iterator) Record() *sam.Record { return i.rec }
 
 // Close releases the underlying Reader.
 func (i *Iterator) Close() error {
-	i.r.SetChunk(nil)
+	if err := i.r.SetChunk(nil); err != nil {
```

Comment:
This isn't necessary. `SetChunk(nil)` always returns `nil`.

Reply:
Ok. I prefer having a simple check just to be sure. But it's your call.